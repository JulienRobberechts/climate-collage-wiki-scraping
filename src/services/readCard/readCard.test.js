/**
 * @jest-environment node
 */
const { readCard } = require('./readCard');

describe('readCard', () => {
  it('read carte_1', async () => {
    const cardNumber = 1;
    const pageData = await readCard(cardNumber);
    expect(pageData).toStrictEqual({
      cardId: 4,
      cardNum: 1,
      url: "/wiki/index.php?title=Fr-fr_adulte_carte_1_activit%C3%A9s_humaines",
      title: "Fr-fr adulte carte 1 activités humaines",
      shortTitle: "Activités humaines",
      cardBatch: 1,
      causes: [
      ],
      consequences: [
        "/wiki/index.php?title=Fr-fr_adulte_carte_2_industrie",
        "/wiki/index.php?title=Fr-fr_adulte_carte_3_b%C3%A2timent",
        "/wiki/index.php?title=Fr-fr_adulte_carte_4_transport",
        "/wiki/index.php?title=Fr-fr_adulte_carte_8_agriculture",
      ],
    });
  });
  it('read carte_2', async () => {
    const cardNumber = 2;
    const pageData = await readCard(cardNumber);
    expect(pageData).toStrictEqual({
      cardId: 12,
      cardNum: 2,
      url: "/wiki/index.php?title=Fr-fr_adulte_carte_2_industrie",
      title: "Fr-fr adulte carte 2 industrie",
      shortTitle: "Industrie",
      cardBatch: 2,
      causes: [
        "/wiki/index.php?title=Fr-fr_adulte_carte_1_activit%C3%A9s_humaines",
      ],
      consequences: [
        "/wiki/index.php?title=Fr-fr_adulte_carte_5_%C3%A9nergies_fossiles",
        "/wiki/index.php?title=Fr-fr_adulte_carte_10_a%C3%A9rosols",
      ],
    });
  });
});
