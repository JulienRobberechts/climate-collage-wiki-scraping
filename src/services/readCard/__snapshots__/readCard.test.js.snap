// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readCard read carte_9 1`] = `
Object {
  "cardNum": 9,
  "causes": Array [
    "/wiki/index.php?title=Fr-fr_adulte_carte_8_agriculture",
    "/wiki/index.php?title=Fr-fr_adulte_carte_41_permafrost",
    "/wiki/index.php?title=Fr-fr_adulte_carte_42_hydrates_de_m%C3%A9thane",
  ],
  "consequences": Array [
    "/wiki/index.php?title=Fr-fr_adulte_carte_13_effet_de_serre_additionnel",
  ],
  "img": Object {
    "url": "https://fresqueduclimat.org/wiki/images/3/38/Fr-fr_adulte_carte_9_recto.png",
  },
  "title": "Autres GES",
  "wikiId": 26,
  "wikiInternalName": "Fr-fr adulte carte 9 autres ges",
  "wikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_9_autres_ges",
}
`;

exports[`readCard read carte_14 1`] = `
Object {
  "cardNum": 14,
  "causes": Array [
    "/wiki/index.php?title=Fr-fr_adulte_carte_15_for%C3%A7age_radiatif",
  ],
  "consequences": Array [
    "/wiki/index.php?title=Fr-fr_adulte_carte_16_fonte_des_glaciers",
    "/wiki/index.php?title=Fr-fr_adulte_carte_17_hausse_temp%C3%A9rature_eau",
    "/wiki/index.php?title=Fr-fr_adulte_carte_18_fonte_de_la_banquise",
    "/wiki/index.php?title=Fr-fr_adulte_carte_19_fonte_des_calottes_glaciaires",
    "/wiki/index.php?title=Fr-fr_adulte_carte_21_hausse_temp%C3%A9rature",
  ],
  "img": Object {
    "url": "https://fresqueduclimat.org/wiki/images/f/fe/Fr-fr_adulte_carte_14_recto.png",
  },
  "title": "Bilan énergétique",
  "wikiId": 36,
  "wikiInternalName": "Fr-fr adulte carte 14 bilan énergétique",
  "wikiUrl": "/wiki/index.php?title=Fr-fr_adulte_carte_14_bilan_%C3%A9nerg%C3%A9tique",
}
`;
